tests/std/tests/GH_002989_nothrow_unwrappable/test.cpp

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <filesystem>
+#include <list>
+#include <type_traits>
+#include <vector>
+#include <xutility>
+
+#if _HAS_CXX20
+#include <ranges>
+#endif
+
+using namespace std;
+
+#if _HAS_CXX17
+using filesystem::path;
+#endif
+
+#define STATIC_ASSERT(...) static_assert(__VA_ARGS__, #__VA_ARGS__)
+
+struct Predicate {
+    template <class T>
+    bool operator()(const T&) const {
+        return true;
+    }
+};
+
+template <class It, bool CopyUnwrapNothrow = true>
+void do_single_test() {
+    // !a || b is equivalent to a => b (a implies b)
+    // This is written this way to avoid `if constexpr` in C++14 mode.
+    STATIC_ASSERT(!_Unwrappable_v<It> || _Is_nothrow_unwrappable_v<It>);
+    STATIC_ASSERT(!_Unwrappable_v<It> || _Is_nothrow_unwrappable_v<It&&>);
+    STATIC_ASSERT(noexcept(_Get_unwrapped(declval<It>())));
+
+    STATIC_ASSERT(!_Unwrappable_v<It> || _Is_nothrow_unwrappable_v<const It&> == CopyUnwrapNothrow);

```

Comment:
`==` has lower precedence than `||`. I think you want:
```suggestion
    STATIC_ASSERT(!_Unwrappable_v<It> || (_Is_nothrow_unwrappable_v<const It&> == CopyUnwrapNothrow) );
```